### DISCUSSION TITLE
Export template format to call information from device such as interfaces

### DISCUSSION THREAD

**Discussion by TimY8976** [netbox-community/netbox#11733] [Q&A]
Hi All,
I'm working on setting up export templates to generate config for devices. I'm having a bit of an issue when trying to call some device specific information such as interface. For instance, if i set device for query set and pull data from config context, that works. But I also want to pull loopback interface IP, i keep on getting attribute error. 

Wondering how i can map interface data and device data in same template?

Thanks

**Comment by candlerb** [comment]
If you show what you've done, maybe someone can spot the error.

Export templates aren't *really* intended for generating device configs (they are intended to iterate over a *list* of devices), but it might be possible to (ab)use them for this purpose.

The objects which you get from the queryset are of whatever object type you requested to export, but related objects can be found via accessors, e.g. 'object.interfaces'

As an example, here's what I use for a custom link (which should be similar to export template) to link to the management IP address:

```
# Link text
{% if object.interfaces.filter(mgmt_only=True).count() == 1 and object.interfaces.get(mgmt_only=True).ip_addresses.count() == 1 %}Open IPMI{% endif %}

# Link URL
https://{{ object.interfaces.get(mgmt_only=True).ip_addresses.first().address.ip }}/
```

If you want to experiment with this, [nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/) is a good place to play.

**Comment by TimY8976** [comment]
Thank you, this pointed us in the right direction on how to pull the Lo0. We were attempting something like the below. We are attempting to create small export templates for pieces of code long term. It seems like based on your example above, putting more information into config context vs pre-built areas would be more flexible as you cant always tie a value from the object (VRF, or IPAM info)

{% for device in queryset %}
{% for interface in device.interfaces.Lo0()  %}
interface_name {{ interface.name }} 
{% for interface in interfaces %}
set protocols bgp group GROUP local-address {{ interfaces.device.Lo0 }}

**Reply by candlerb** [reply]
> ```
> {% for interface in device.interfaces.Lo0() %}
> ```

I wouldn't expect that to work (although if it does, then great).  I would have thought something like:

```
{% for interface in device.interfaces.filter(name="Lo0") %}
```

Note that since there can only be one interface with a given name, this will iterate only zero or one times.

> ```
> {% for interface in interfaces %}
> ```

I'm not sure what you're trying to do there: maybe `{% for interface in device.interfaces %}` ?
